filtered-resource-service/README.md

```diff
@@ -62,26 +62,44 @@ See filtered-resource-service/doc/filtered-resource-service.drawio for the sourc
 The main route of web requests - and subsequent websocket messages - through the service is shown in the above diagram. 
 The green boxes are client web requests, red boxes are kafka topics, and red storage is redis persistence.
 
-### Token-Offset System
+### Token-Offset Actor System
 
 The top-right box is the token-offset-system and its associated worker actors. 
 This performs the task of persisting and retrieving offsets for tokens. 
 We use an Akka Actor System to process both requests from the client and messages from kafka. 
-In the former case, the offset is either looked-up from redis or told to the actor from kafka. 
-In the latter case, the offset is persisted to redis and then told to all running actors for that given token.
+The first diagram below shows what happens when a client connects to the service before the request has been processed by Palisade.
+The second diagram below shows what happens when a client connects to the service after the request has been processed by Palisade.
 
-### Token-Error System
+#### Early Client Connection
+![Filtered-Resource Service Early Client Connection Diagram](doc/Filtered-Resource-Offset-Early-Client.png)
+
+If the client connects to the service before the request has been processed by Palisade an Akka Actor Worker is spawned, which deals with the incoming offset value.
+The worker will check Redis to check if there are any values present for the token, this will return an empty value.
+The service will then read messages from the `masked-resource-offset` Kafka topic until it finds one for the given token.
+The service will tell the Akka Actor Worker the offset value which is then used by the service to read the resources from the `masked-resource` Kafka topic.
+
+#### Late Client Connection
+![Filtered-Resource Service Late Client Connection Diagram](doc/Filtered-Resource-Offset-Late-Client.png)
+
+If the client connects to the service after Palisade has finished processing the request, the information from the `masked-resource-offset` Kafka topic is stored in Redis.
+The `GET` from the client triggers the service to spawn an Akka Actor Worker.
+The worker will then check Redis for any values that match the incoming token value.
```

Comment:
```suggestion
The worker will then check Redis for any values that match the incoming token value (where it will find the offset since the client connected late as described).
```

Reply:
Done